ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -416,4 +424,32 @@ public static String getOperationOrBlobIdFromUri(RestRequest restRequest, RestUt
     }
     return subResource;
   }
+
+  /**
+   * Fetch time in ms for the {@code dateString} passed in, since epoch
+   * @param dataString the String representation of the date that needs to be parsed
+   * @return Time in ms since epoch. Note http time is kept in Seconds so last three digits will be
+   *         000. Returns null in case of parse exception
+   */
+  public static Long getTimeFromDateString(String dataString) {
+    SimpleDateFormat dateFormatter = new SimpleDateFormat(HTTP_DATE_FORMAT, Locale.US);
```

Comment:
Is the Locale always US? What about deployments not in the US?

I don't think we should be parsing this. Try the suggestion that I have given above.